puts/harness/include/tlspuffin/put.h

```diff
@@ -0,0 +1,155 @@
+#ifndef TLSPUFFIN_PUT_H
+#define TLSPUFFIN_PUT_H
+
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+// FIXME: C/Rust duplication of type definitions
+//
+//     All the types related to the descriptor are equivalent of the Rust
+//     structs and enums. To avoid discrepancy, we could generate a header from
+//     Rust using `cbindgen` or use the C definition in Rust using `bindgen`.
+//
+//     Another solution would be to expose functions to retrieve config
+//     informations from Rust. This has the advantage to completely hide the
+//     rust structs at the price of a small performance cost each time one of
+//     these functions is called.
+
+typedef void *RESULT;
+
+typedef enum
+{
+    V1_3,
+    V1_2
+} TLS_VERSION;
+
+typedef enum
+{
+    CLIENT,
+    SERVER
+} AGENT_TYPE;
+
+typedef struct
+{
+    const uint8_t *const bytes;
+    const size_t length;
+} PEM;
+
+typedef enum
+{
+    RESULT_OK,
+    RESULT_IO_WOULD_BLOCK,
+    RESULT_ERROR_FATAL,
+    RESULT_ERROR_OTHER
+} RESULT_CODE;
+
+typedef struct
+{
+    uint8_t name;
+    AGENT_TYPE type;
+    TLS_VERSION tls_version;
+    bool client_authentication;
+    bool server_authentication;
+    const PEM *cert;
+    const PEM *pkey;
+
+    const PEM *const *const store;
+    const size_t store_length;
```

Comment:
Which store is this?

Reply:
This is the agent's certificates store. The decision of what to include is left to the Rust part in tlspuffin. In terms of what exists today in the code:
- the server (ALICE) store includes BOB and EVE certificates
- the client (BOB) store includes ALICE and EVE certificates